src/tinygp/kernels/quasisep.py

```diff
@@ -39,6 +40,8 @@
 from tinygp.solvers.quasisep.core import DiagQSM, StrictLowerTriQSM, SymmQSM
 from tinygp.solvers.quasisep.general import GeneralQSM
 
+eta = 1e-20  # avoid nan
```

Comment:
I don't _love_ this magic number here. I think that the right value probably depends a bit on the datatype (`float32` vs `float64`) and perhaps other considerations. Would it be possible to move this into the class itself and perhaps add some more testing / commenting?

Reply:
Yes, you are correct. It depends on the datatype. I did some tests and now it should be fine with `float32`, but probably will fail with `float16`. I made eta a class parameter, so people can change this based on their desired precision.